tests/rpmbuild.at

```diff
@@ -2961,6 +2961,7 @@ RPMTEST_CLEANUP
 # Check if dynamic spec generation works for main package, too
 AT_SETUP([rpmbuild with dynamic spec generation for main package])
 AT_KEYWORDS([build])
+AT_XFAIL_IF([test $RPM_XFAIL -ne 0])
```

Comment:
AT_XFAIL_IF indicates a known bug, so AIUI it shouldn't be used here. When testing for bad input, just expect the failure code (and error messages) in the test directly.

Reply:
Yes, this is intentional. The idea is that this change is reversed at some point and the test passes again. I should probably open a ticket for 6.0.

Reply:
Guess I should add a test that catches the current failure...

Reply:
Yeah, do that. Yeah I was puzzling here a bit about this because -ba is not a problematic case at all, but thought it could be just down to sleep-deprivation and other disturbances today :sweat_smile:

Reply:
OK, added proper test case and merged the test case changes with the actual implementation to not have a "broken" commit.

Reply:
Ack, that + the added comments make it much clearer now.
Having had a good night's sleep doesn't hurt either.